client/service/src/builder.rs

```diff
@@ -51,7 +51,71 @@ use sysinfo::{get_current_pid, ProcessExt, System, SystemExt};
 use sc_telemetry::{telemetry, SUBSTRATE_INFO};
 use sp_transaction_pool::{TransactionPool, TransactionPoolMaintainer};
 use sp_blockchain;
-use grafana_data_source::{self, record_metrics};
+use prometheus_exporter::{create_gauge, Gauge, U64, F64, Registry, PrometheusError};
+
+struct ServiceMetrics {
+	finality_height: Gauge<U64>,
+	best_height: Gauge<U64>,
+	peers_num: Gauge<U64>,
+	tx_count: Gauge<U64>,
```

Comment:
In case this is indeed a monotonically increasing counter, the convention is to suffix it with `total`, thus `tx_total` or `tx_count_total`, preferring the former.

Reply:
Is the abbreviation necessary here? If not, I would suggest using `transaction` instead of `tx`. Prometheus will compress the data very well on disc, thus no need for brevity here.